core/trino-main/src/main/java/io/trino/sql/planner/SymbolUtils.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Collection;
+import java.util.List;
+
+public class SymbolUtils
+{
+    private SymbolUtils() {}
+
+    public static boolean containsAll(List<Symbol> haystack, Collection<Symbol> needles)
+    {
+        return ImmutableSet.copyOf(haystack).containsAll(needles);
+    }
+
+    public static boolean containsNone(Collection<Symbol> values, Collection<Symbol> testValues)
+    {
+        return values.stream().noneMatch(ImmutableSet.copyOf(testValues)::contains);
```

Comment:
why iterate this one and not the other

Reply:
This method was just moved to a single utility class. It didn't change over the original one :)